serviceregistration/kubernetes/service_registration.go

```diff
@@ -0,0 +1,195 @@
+package kubernetes
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	sr "github.com/hashicorp/vault/serviceregistration"
+	"github.com/hashicorp/vault/serviceregistration/kubernetes/client"
+)
+
+const (
+	// Labels are placed in a pod's metadata.
+	labelVaultVersion = "vault-version"
+	labelActive       = "vault-active"
+	labelSealed       = "vault-sealed"
+	labelPerfStandby  = "vault-perf-standby"
+	labelInitialized  = "vault-initialized"
+
+	// This is the path to where these labels are applied.
+	pathToLabels = "/metadata/labels/"
+)
+
+func NewServiceRegistration(config map[string]string, logger hclog.Logger, state sr.State, _ string) (sr.ServiceRegistration, error) {
+	namespace, err := getRequiredField(logger, config, client.EnvVarKubernetesNamespace, "namespace")
+	if err != nil {
+		return nil, err
+	}
+	podName, err := getRequiredField(logger, config, client.EnvVarKubernetesPodName, "pod_name")
+	if err != nil {
+		return nil, err
+	}
+	return &serviceRegistration{
+		logger:       logger,
+		namespace:    namespace,
+		podName:      podName,
+		initialState: state,
+		retryHandler: &retryHandler{
+			logger:         logger,
+			namespace:      namespace,
+			podName:        podName,
+			patchesToRetry: make([]*client.Patch, 0),
+		},
+	}, nil
+}
+
+type serviceRegistration struct {
+	logger             hclog.Logger
+	namespace, podName string
+	client             *client.Client
+	initialState       sr.State
+	retryHandler       *retryHandler
+}
+
+func (r *serviceRegistration) Run(shutdownCh <-chan struct{}, wait *sync.WaitGroup) error {
+	c, err := client.New(r.logger, shutdownCh)
+	if err != nil {
+		return err
+	}
+	r.client = c
+	r.retryHandler.client = c
+
+	// Verify that the pod exists and our configuration looks good.
+	pod, err := c.GetPod(r.namespace, r.podName)
+	if err != nil {
+		return err
+	}
+
+	// Now to initially label our pod.
+	if pod.Metadata == nil {
+		// This should never happen IRL, just being defensive.
+		return fmt.Errorf("no pod metadata on %+v", pod)
+	}
+	if pod.Metadata.Labels == nil {
+		// Add the labels field, and the labels as part of that one call.
+		// The reason we must take a different approach to adding them is discussed here:
+		// https://stackoverflow.com/questions/57480205/error-while-applying-json-patch-to-kubernetes-custom-resource
+		if err := c.PatchPod(r.namespace, r.podName, &client.Patch{
+			Operation: client.Add,
+			Path:      "/metadata/labels",
+			Value: map[string]string{
+				labelVaultVersion: r.initialState.VaultVersion,
+				labelActive:       toString(r.initialState.IsActive),
```

Comment:
nit: a standard library function exists for "toString":  `strconv.FormatBool()`

Reply:
Replace in https://github.com/hashicorp/vault/pull/8249/commits/e92f7168f5c75884ed67d0371498927f411afc8c